#######################
## Robot Arm Config ###
#######################

robot_arm: "ur3e"

home_joint_states:
  shoulder_pan_joint: 0 
  shoulder_lift_joint: -1.5633965845084568
  elbow_joint: -0.4213316738605499
  wrist_1_joint: -2.73153080562734
  wrist_2_joint: 1.5632756392108362
  wrist_3_joint: 0.0


ready_joint_states:
  shoulder_pan_joint: 0.49431923031806946
  shoulder_lift_joint: -1.220551536684372
  elbow_joint: -1.9489996433258057
  wrist_1_joint: -1.5499402706376095
  wrist_2_joint: 1.5660009384155273
  wrist_3_joint: 0.4943270683288574



arm_group: "ur_manipulator"
base_link: "base_link"
effector_link: "tool0"

eff_ready_pose:  [-0.16, -0.13, 0.5]   # x, y, z for end-effector
eff_ready_orien: [0.70, 0.70, 0.0, 0.0]
fix_orien: [0.70, 0.70, 0.0, 0.0]
work_R: 0.52
work_r: 0.24






##########################
## Robot Gripper Config ##
##########################
robot_gripper: "active_parallel_gripper"

# CAUTION!!! following value needs to be calibrated
g2e_offset: 0.158 #0.168 #0.202 #0.168

###################
## Camera Config ##
###################

camera_pose: [-0.385, 0.028, 0.738000]
camera_orien: [180.0, 0.000000, 89.500000] # Euler system in degree.
# CAUTION!!! following values need to be calibrations using hand-eye-calibration
# camera_setup: "eye-on-hand"
# eff2camera_pos:
#   pose: [-0.12, -0.12, -0.035]
#   orien: [7.93796318e-05, -4.68563589e-04, -9.23782058e-01,  3.82918377e-01] ## target will be [0, 1, 0, 0]

load_intrinsic: "from_camera"

crop_resol: [1280, 700] # [1200, 700]
crop_start_x: 0 #50
crop_start_y: 20 #50

# ####################
# ## MoveIt Setting ##
# ####################

# ros_sleep_time: 0.2
# max_vel_scale: 0.3
# max_acc_scale: 0.1
# plan_pipe_id: 'pilz_industrial_motion_planner'
# planner_id: 'PTP'

############################
## Pick and Place Setting ##
############################

pick_raise_offset: 0.03 
place_raise_offset: 0.03
crop_scale: 0.5